pyad/forward_diff.py

```diff
@@ -108,20 +199,84 @@ def __init__(self, name, value):
 
 # Elementary operations of a single variable. They all use chain rule.
 def _elementary_op(obj, fn, deriv_fn):
+    '''
+    A generic framework to allow for the chain rule of other elementary functions 
+    taken from the numpy module.
+
+    Parameters
+    ----------
+    obj : Class
+        The Variable or Tensor which the elementary function is being differentiated at
+
+    fn : np.function
```

Comment:
I would probably just say that these show be general python functions rather than numpy functions specifically, since there are some usages of this with lambdas